docs/DeveloperGuide.md

```diff
@@ -646,6 +646,93 @@ testers are expected to do more *exploratory* testing.
+       
+    1. Other incorrect `edit-doctor` commands to try: `edit-doctor 1`, `...` (where no field to edit is provided)<br>
+       Expected: An error message informing that at least one field to edit must be provided.
+
+1. Editing results in duplicated doctors
+
+    1. Prerequisites: The edited name of the doctor has the same `Name` as an existing doctor in the doctor records.
+
+    1. Test case: `edit-doctor 1 n/Dr Who`<br>
+       Expected: The name of doctor 1 is not edited. Error details are shown in the status message. Status bar remains the same.
+
+### Edit an appointment
+
+1. Editing an appointment
+
+    1. Prerequisites: The appointment to edited must already exist in the appointment schedule.
+
+    1. Test case: `edit-appt 1 pt/2`<br>
+       Expected: The patient of the first appointment in the appointment schedule is changed to the second patient in the patient list, and the details of the appointments are displayed in the status message.
+
+    1. Test case: `edit-appt 2 at/2021-05-08 09:00`<br>
+       Expected: The starting time of the second appointment in the appointment schedule is changed to 2021-05-08 09:00. If the starting time is the same or after the end time of the appointment, and error message will be shown indicating the timeslot end date and time must be strictly after the start date and time.
+
+    1. Other incorrect `edit-appt` commands to try: `edit-appt 1 at/2021-*-08 09:00 `, `...` (where the starting time format provided is invalid)<br>
+       Expected: An error message informing 'Invalid Date Time Format!' is shown.
+
+1. Editing results in conflicting appointments
+
+    1. Prerequisites: The edited appointment has conflicting patient or doctor timeslots with an existing appointment in the appointment schedule.
+    1. Test case: `edit-appt 1 to/2022-06-08 09:00`<br>
+       Expected: The end time of the first appointment in the appointment schedule is not changed. Error details are shown in the status message. Status bar remains the same.
+       
+### Find patients by search fields
+
+1. Finding a patient
+
+    1. Prerequisites: The patient to be found must exits in the patient list.
+
+    1. Test case: `find-patient John`<br>
+       Expected: The patient with name John is found and listed.
+
+   1. Test case: `find-patient 84511556`<br>
+      Expected: 0 patient is found. This is because `find-patient` command only search for keywords in the patient's name field.
+      
+Note: This test case can be similarly performed for doctors in the doctor records by replacing `find-patient` with `find-doctor`.
+       
+### Find doctors by search fields
+
+This test case for finding a patient can be similarly performed here for doctors in the doctor records by replacing `find-patient` with `find-doctor`.
+
+### Delete a doctor
+
+This test case for deleting a patient can be similarly performed here for doctors in the doctor records by replacing `delete-patient` with `delete-doctor`.
+
```

Comment:
why are these here? these are supposed to be included with the patient variant of the command, not as a separate command as per below:
> this test case for finding a patient can be similarly performed here